src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -64,19 +65,34 @@ public ObjectResult(object value)
         {
             var incomingAcceptHeader = HeaderParsingHelpers.GetAcceptHeaders(
                                                 formatterContext.ActionContext.HttpContext.Request.Accept);
-            var sortedAcceptHeaders = SortMediaTypeWithQualityHeaderValues(incomingAcceptHeader)
+            var sortedAcceptHeaderMediaTypes = SortMediaTypeWithQualityHeaderValues(incomingAcceptHeader)
                                         .Where(header => header.Quality != HttpHeaderUtilitites.NoMatch)
                                         .ToArray();
 
+            var options = formatterContext.ActionContext.HttpContext
+                                                        .RequestServices
+                                                        .GetRequiredService<IOptions<MvcOptions>>()
+                                                        .Options;
+
             IOutputFormatter selectedFormatter = null;
 
+            bool ignoreAcceptHeader = false;
```

Comment:
I tried doing this where its backed by a static field but this is causing issue in in-memory tests as they do not seem to be running in sequence causing the tests to randomly succeed/fail.